webapp/channels/src/packages/mattermost-redux/src/actions/shared_channels.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {RemoteCluster} from '@mattermost/types/remote_clusters';
+import type {GlobalState} from '@mattermost/types/store';
+
+import {Client4} from 'mattermost-redux/client';
+import type {ActionFuncAsync} from 'mattermost-redux/types/actions';
+
+import {ActionTypes} from '../reducers/entities/remote_clusters';
+
+export function receivedRemoteClusters(remoteClusters: RemoteCluster[]) {
+    return {
+        type: ActionTypes.RECEIVED_REMOTE_CLUSTERS,
+        data: remoteClusters,
+    };
+}
+
+export function fetchRemoteClusters(): ActionFuncAsync<RemoteCluster[]> {
+    return async (dispatch: any, getState: () => GlobalState) => {
+        // Check if we already have the data in the Redux store
+        const state = getState();
+        const remoteClusters = state.entities?.remoteClusters;
+
+        // If we already have the data, no need to fetch it again
+        if (remoteClusters && Object.keys(remoteClusters).length > 0) {
+            return {data: Object.values(remoteClusters)};
+        }
```

Comment:
Remote clusters could change through the lifecycle of the webapp, and I don't think we have WS messages to push those changes and trigger a refetch. Should we remove this check then so the webapp checks the remotes for changes when necessary?

I'm fine leaving it as is too and trying to implement a WS message for the backend to notify the webapp as a known improvement. 0/5

Reply:
re:"Remote clusters could change through the lifecycle of the webapp," - while technically true, establishing connections with more remote clusters is a heavy duty op. I believe, so I think that using thr cache in most cases should be ok.